typings/index.d.ts

```diff
@@ -0,0 +1,144 @@
+        giveaway?: string;
+        giveawayEnded?: string;
+        inviteToParticipate?: string;
+        timeRemaining?: string;
+        winMessage?: string;
+        embedFooter?: string;
+        noWinner?: string;
+        winners?: string;
+        endedAt?: string;
+        hostedBy?: string;
+        units?: {
+            seconds?: string;
+            minutes?: string;
+            hours?: string;
+            days?: string;
+            pluralS?: false;
+        };
+    }
+    interface GiveawaysManagerEvents {
+        giveawayEnded: [Giveaway, GuildMember[]];
+        giveawayReactionAdded: [Giveaway, GuildMember, MessageReaction];
+        giveawayReactionRemoved: [Giveaway, GuildMember, MessageReaction];
+    }
+    class Giveaway extends EventEmitter {
+        constructor(manager: GiveawaysManager, options: GiveawayData)
+
+        public botsCanWin: boolean;
+        readonly channel: TextChannel;
+        public channelID: Snowflake;
+        public client: Client;
+        readonly content: string;
+        public data: GiveawayData;
+        public embedColor: ColorResolvable;
+        public embedColorEnd: ColorResolvable;
+        public endAt: number;
+        public ended: boolean;
+        public exemptPermissions: PermissionResolvable[];
+        readonly giveawayDuration: number;
+        public guildID: Snowflake;
+        public hostedBy: string | null;
+        public manager: GiveawaysManager;
+        public message: Message | null;
+        public messageID: Snowflake | null;
+        public messages: GiveawaysMessages;
+        public options: GiveawayData;
+        public prize: string;
+        readonly remainingTime: number;
+        public startAt: number;
+        public winnerCount: number;
+
+        public exemptMembers(): boolean;
+        public edit(options: GiveawayEditOptions): Promise<Giveaway>;
+        public end(): Promise<GuildMember[]>
+        // @ts-ignore-next-line
+        public async fetchMessages(): Promise<Message>
+        public reroll(options: GiveawayRerollOptions): Promise<GuildMember[]>
+        // @ts-ignore-next-line
+        public async roll(winnerCount?: number): Promise<GuildMember | GuildMember[]>;
+    }
+    interface GiveawayEditOptions {
+        newWinnerCount?: number;
+        newPrize?: string;
+        addTime?: number;
+        setEndTimestamp?: number;
+    }
+    interface GiveawayRerollOptions {
+        winnerCount?: number | null;
+        messages?: {
+            congrat?: string;
+            error?: string;
+        };
+    }
+    interface GiveawayData {
+        startAt: number;
+        endAt: number;
+        winnerCount: number;
+        messages: GiveawaysMessages;
+        ended: boolean;
+        prize: string;
+        channelID: Snowflake;
+        guildID: Snowflake;
+        messageID?: Snowflake | null;
+        reaction?: string;
+        exemptPermissions?: PermissionResolvable[];
+        exemptMembers?: () => boolean;
```

Comment:
I don't know this is valid in TypeScript but I think it is
```suggestion
        exemptMembers?: (member: GuildMember) => boolean;
```